File: VB/Custom_Extract_Data_Source_Wizard/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardCommon.DataSourceWizard
Imports DevExpress.DashboardWin
Imports DevExpress.DashboardWin.DataSourceWizard
Imports DevExpress.DataAccess.UI.Wizard
Imports DevExpress.XtraEditors
Imports System

Namespace Custom_Extract_Data_Source_Wizard
	Partial Public Class Form1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btn_RunWizard_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btn_RunWizard.Click
			Dim optionsContext As New EditExtractOptionsContext(Me.GetActiveLookAndFeel(), Me, dashboardViewer1.Dashboard.DataSources)
			optionsContext.WizardCustomization = New CustomExtractDataSourceWizardCustomizationService()
			ExtractDataSourceUIHelper.EditExtractOptions(New DashboardExtractDataSource() With {.FileName = "test.dat"}, optionsContext)
		End Sub
	End Class

	Public Class CustomExtractDataSourceWizardCustomizationService
		Implements IExtractDataSourceWizardCustomization

		Public Sub CustomizeDataSourceWizard(ByVal customization As IWizardCustomization(Of ExtractDataSourceModel)) Implements IExtractDataSourceWizardCustomization.CustomizeDataSourceWizard
			customization.RegisterPageView(Of ICreateExtractDataSourcePageView, CustomConfigureExtractDataSourcePageView)()
		End Sub

	End Class

	Public Class CustomConfigureExtractDataSourcePageView
		Inherits CreateExtractDataSourcePageView

		Public Sub New()
			MyBase.New()
			Me.checkEditExistingDataSource.Enabled = False

		End Sub

		Protected Overrides Sub OnCheckEditExistingDataSourceCheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
			MyBase.OnCheckEditExistingDataSourceCheckedChanged(sender, e)
			TextEditFileName.Enabled = False
		End Sub

	End Class
End Namespace

File: CS/Custom_Extract_Data_Source_Wizard/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardCommon.DataSourceWizard;
using DevExpress.DashboardWin;
using DevExpress.DashboardWin.DataSourceWizard;
using DevExpress.DataAccess.UI.Wizard;
using DevExpress.XtraEditors;
using System;

namespace Custom_Extract_Data_Source_Wizard {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
        }

        private void btn_RunWizard_Click(object sender, EventArgs e) {
            EditExtractOptionsContext optionsContext = new EditExtractOptionsContext(this.GetActiveLookAndFeel(), this, dashboardViewer1.Dashboard.DataSources);
            optionsContext.WizardCustomization = new CustomExtractDataSourceWizardCustomizationService();
            ExtractDataSourceUIHelper.EditExtractOptions(new DashboardExtractDataSource() { FileName = "test.dat" }, optionsContext);
        }
    }

    public class CustomExtractDataSourceWizardCustomizationService : IExtractDataSourceWizardCustomization {
        public void CustomizeDataSourceWizard(IWizardCustomization<ExtractDataSourceModel> customization) {
            customization.RegisterPageView<ICreateExtractDataSourcePageView, CustomConfigureExtractDataSourcePageView>();
        }
    }

    public class CustomConfigureExtractDataSourcePageView : CreateExtractDataSourcePageView {
        public CustomConfigureExtractDataSourcePageView() : base() {                     
            this.checkEditExistingDataSource.Enabled = false;            
        }

        protected override void OnCheckEditExistingDataSourceCheckedChanged(object sender, EventArgs e) {
            base.OnCheckEditExistingDataSourceCheckedChanged(sender, e);
            TextEditFileName.Enabled = false;
        }
    }
}
